pkg/sql/pgwire/types.go

```diff
@@ -705,14 +707,27 @@ func decodeOidDatum(id oid.Oid, code formatCode, b []byte) (parser.Datum, error)
 	// Types with identical text/binary handling.
 	switch id {
 	case oid.T_text, oid.T_varchar:
+		if err := validateStringBytes(b); err != nil {
+			return nil, err
+		}
 		return parser.NewDString(string(b)), nil
 	case oid.T_name:
+		if err := validateStringBytes(b); err != nil {
+			return nil, err
+		}
 		return parser.NewDName(string(b)), nil
 	default:
 		return nil, errors.Errorf("unsupported OID %v with format code %s", id, code)
 	}
 }
 
+func validateStringBytes(b []byte) error {
```

Comment:
Should we add a comment on here stating which types need to be valid utf-8 and which ones don't?